### PR TITLE
Refactor access_token_sign_in_required and AYRUser in preparation for auth token refresh flow

### PR DESCRIPTION
**Pull request by anthonyhashemi**
## Changes in this PR

- Refactor access_token_sign_in_required and AYRUser in preparation for auth token refresh flow

## JIRA ticket

https://national-archives.atlassian.net/browse/AYR-420

### CODE REVIEW

File: app/main/authorize/access_token_sign_in_required.py [hunk 1]
```diff
@@ -57,3 +59,26 @@ def decorated_view(*args, **kwargs):
     decorated_view.access_token_sign_in_required = True
 
     return decorated_view
+
+
+def _validate_access_token():
+    valid_access_token = False
```
Comment (anthonyhashemi) [inline app/main/authorize/access_token_sign_in_required.py]: `is_valid_access_token`

Comment (anthonyhashemi) [reply app/main/authorize/access_token_sign_in_required.py]: have done

File: app/main/authorize/access_token_sign_in_required.py [hunk 2]
```diff
@@ -57,3 +59,26 @@ def decorated_view(*args, **kwargs):
     decorated_view.access_token_sign_in_required = True
 
     return decorated_view
+
+
+def _validate_access_token():
+    is_valid_access_token = False
+
+    keycloak_openid = keycloak.KeycloakOpenID(
```
Comment (anthonyhashemi) [inline app/main/authorize/access_token_sign_in_required.py]: take a look at generalising a global object. not 100% sure if I want to but will look into .

Comment (anthonyhashemi) [reply app/main/authorize/access_token_sign_in_required.py]: done

**Comment by pgandhizaizi** [comment]
like the extraction into config helpers

**Review by pgandhizaizi — APPROVED** [review]
thank you for providing detailed overview of changes in PR in a huddle, really helpful to understand the keycloak changes,

also like the extraction in to flask_config_helpers

**Pull request merged**